### PR TITLE
Fix encoding of empty parameters

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #50 

This PR solves the issue by setting the value of an empty parameter to a default value of the correct type, as described in the parent issue. By adding back the tests that previously didnt pass, we can see that the issue of encoding and decoding empty parameters should now be resolved

## Review guidelines

Estimated Time of Review: 5 minutes

### CODE REVIEW

File: python/source/clproto/bind_clproto.cpp [hunk 1]
```diff
@@ -27,76 +27,86 @@ inline py::bytes encode_bytes(const T& object) {
...
+  auto param = decode<Parameter<T>>(msg);
+  if (param.is_empty()) {
+    return PyParameter(param.get_name(), param.get_parameter_type());
+  } else {
+    return PyParameter(param.get_name(), py::cast(param.get_value()), param.get_parameter_type());
+  }
+}
+
 py::object decode_parameter(const std::string& msg) {
   py::object PyParameter = py::module_::import("state_representation").attr("Parameter");
```
Comment (eeberhard) [inline python/source/clproto/bind_clproto.cpp:79]: This line is not needed anymore now that you have the helper function
```suggestion
```

**Review by eeberhard — COMMENTED** [review]
Nice work. The change to the encoding logic is very simple and sensible, and with the templated test cases and additional helper functions that you introduced it appears very solid to me.

Review time was 15-20 minutes just for double checking each case, the test parameterization and python bindings

**Comment by domire8** [comment]
> Review time was 15-20 minutes just for double checking each case, the test parameterization and python bindings

I initially forgot that the Python bindings needed changes as well, the 5 minutes were wrong as a consequence of this...

**Review by eeberhard — APPROVED** [review]
